src/index.js

```diff
@@ -1,225 +1,64 @@
-
-      // If the CAR is existing and not completed, just move the stream to the last analyzed block
-      if (existingCar) {
-        indexer.reader.seek(existingCar.currentPosition - indexer.reader.pos)
-      } else {
-        // Store the initial information of the CAR
-        await writeDynamoItem(true, carsTable, primaryKeys.cars, carId, {
-          bucket: bucketName,
-          bucketRegion: bucketRegion,
-          key,
-          /* c8 ignore next */
-          createdAt: now || new Date().toISOString(),
-          roots: Array.from(new Set(indexer.roots.map(r => r.toString()))),
-          version: indexer.version,
-          fileSize: indexer.length,
-          currentPosition: indexer.reader.pos,
-          completed: false
-        }, carId)
-      }
-
-      let blockCount = 0
-      // For each block in the indexer (which holds the start and end block)
-      await forEach(
-        indexer,
-        async function (block) {
-          blockCount++
-
-          // Show CAR progress
-          logger.debug(
-            {
-              car: carId,
-              elapsed: elapsed(start),
-              progress: {
-                records: { current: currentCarIndex, total: totalCars },
-                car: { position: indexer.position, length: indexer.length }
-              }
-            },
-            `Analyzing CID ${block.cid}`
-          )
-
-          const existingBlock = await readDynamoItem(blocksTable, primaryKeys.blocks, cidToKey(block.cid), carId)
-          if (existingBlock) {
-            return
-          }
-          /*
-          Note that when DECODE_BLOCKS env variable is unset
-          block.data is always undefined and therefore no decoding is performed.
-        */
-          await storeNewBlock(
-            carId,
-            codecs[block.cid.code]?.label ?? 'unsupported',
-            block,
-            carId,
-            block.data ? decodeBlock(block, carId) : undefined
-          )
-
-          // Update the information about the CAR analysis progress
-          await updateCarStatus(carId, block, carId)
-        },
-        concurrency
-      )
-
-      // Mark the CAR as completed and notify to SQS
-      await writeDynamoItem(false, carsTable, 'path', carId, {
-        currentPosition: indexer.length,
-        completed: true,
-        durationTime: skipDurations ? 0 : elapsed(partialStart)
-      }, carId)
-
-      await publishToSQS(notificationsQueue, carId, carId)
-
-      logger.info({
-        car: carId,
-        ...stats,
-        blocks: blockCount,
-        duration: elapsed(partialStart),
-        resumed: Boolean(existingCar)
-      }, `Indexing complete: ${carId}`)
+  try {
+    try {
+      logger.debug('Indexing CARs progress')
 
+      const carLogger = logger.child({ car: carId })
+      await storeCar({ id: carId, skipExists, decodeBlocks, logger: carLogger })
+    } finally {
       telemetry.flush()
```

Comment:
Why is this in a separate try block?

Reply:
`telemetry.flush` must be called even on throwing error

Reply:
Can it not be a finally in the outer try/catch?

Reply:
in that case, `throw err` at line 57 would be ignored

Reply:
ok